channels/ON-CHAIN.md

```diff
@@ -0,0 +1,372 @@
+  initiator_final := 0.0
+```
+
+
+
+#### Requirements
+
+This transaction MUST have valid signatures of all involved parties.
+
+After this transaction has been included in a block, the channel MUST be
+considered closed and allow no further modifications.
+
+
+### `channel_close_solo`
+
+In order to close a channel unilaterally, a user has to send a
+`channel_close_solo` transaction. This is only necessary if one peer stops
+responding but can also be used by an malicious peer trying to close a channel
+with a state that hasn't been agreed on by all participants.
+
+With the inclusion of this transaction on chain, the timer, during which
+disputes in the form of `channel_slash` will be considered, is started.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| amount               | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`:
+- `from`
+- `ttl`
+- `nonce`
+- `fee`
+
+
+### `channel_slash`
+
+If a malicious party sent a `channel_close_solo` with an outdated state, the
+honest party has the opportunity to issue a `channel_slash` transaction. This
+transaction needs to include a state signed by all peers with a higher sequence
+number.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| amount               | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`:
+- `ttl`
+- `nonce`
+- `fee`
+
+
+#### Requirements
+
+MUST be signed using private key corresponding to the public key `from`.
+
+### `channel_settle`
+
+The settlement transaction is the last one in the lifecycle of a channel, but
+only required if the parties involved did not manage to cooperate when trying
+to close the channel. It has to be issued after all possible disputes are
+resolved and then redistributes the locked funds.
+
+The `channel_settle` MUST only be included in a block if:
+
+- a `channel_close_solo` transaction was published and has expired, i.e.
+`blockheight(top) - blockheight(channel_close_solo_tx) >= lock_period`
+- there are no outstanding `channel_slash` transactions, which means that
+  either none were published or the most recent one expired
+
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| initiator_amount     | 32 |
```

Comment:
I understand both initiator_amount and responder_amount are meant not to be trusted but to be verified with though set on the channel on the state tree by previous channel_close_solo and channel_slash

Reply:
Yes, there might have been a single close and numerous slashes, this settles it and indeed those amounts are not to be trusted but to be verified